pkg/azuredisk/controllerserver.go

```diff
@@ -1092,22 +1092,30 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	}
 	mc := metrics.NewMetricContext(consts.AzureDiskCSIDriverName, metricsRequest, d.cloud.ResourceGroup, d.cloud.SubscriptionID, d.Name)
 	isOperationSucceeded := false
+	isOperationInProgress := false
 	defer func() {
-		mc.ObserveOperationWithResult(isOperationSucceeded, consts.SourceResourceID, sourceVolumeID, consts.SnapshotName, snapshotName)
+		if !isOperationInProgress {
+			mc.ObserveOperationWithResult(isOperationSucceeded, consts.SourceResourceID, sourceVolumeID, consts.SnapshotName, snapshotName)
+			klog.V(2).Infof("create snapshot(%s) under rg(%s) region(%s) operation result", snapshotName, resourceGroup, d.cloud.Location)
```

Comment:
pls remove this redundant log since `mc.ObserveOperationWithResult` would already generate more detailed logs